crates/verify/src/bytecode.rs

```diff
@@ -0,0 +1,662 @@
+            eyre::bail!("Contract name mismatch");
+        }
+
+        // Get the constructor args from etherscan
+        let constructor_args = if let Some(args) = source_code.items.first() {
+            args.constructor_arguments.clone()
+        } else {
+            eyre::bail!("No constructor arguments found for contract at address {}", self.address);
+        };
+
+        // Get user provided constructor args
+        let provided_constructor_args = if let Some(args) = self.constructor_args.to_owned() {
+            args
+        } else if let Some(path) = self.constructor_args_path.to_owned() {
+            // Read from file
+            let res = read_constructor_args_file(path)?;
+            // Convert res to Bytes
+            res.join("")
+        } else {
+            constructor_args.to_string()
+        };
+
+        // Constructor args mismatch
+        if provided_constructor_args != constructor_args.to_string() && !self.json {
+            println!(
+                "{}",
+                Paint::red("The provider constructor args do not match the constructor args from etherscan. This will result in a mismatch - Using the args from etherscan").bold(),
+            );
+        }
+
+        // Get creation tx hash
+        let creation_data = etherscan.contract_creation_data(self.address).await?;
+
+        let mut transaction = provider
+            .get_transaction_by_hash(creation_data.transaction_hash)
+            .await
+            .or_else(|_| eyre::bail!("Couldn't fetch transaction from RPC"))?;
+        let receipt = provider
+            .get_transaction_receipt(creation_data.transaction_hash)
+            .await
+            .or_else(|_| eyre::bail!("Couldn't fetch transacrion receipt from RPC"))?;
+
+        let receipt = if let Some(receipt) = receipt {
+            receipt
+        } else {
+            eyre::bail!(
+                "Receipt not found for transaction hash {}",
+                creation_data.transaction_hash
+            );
+        };
+        // Extract creation code
+        let maybe_creation_code = if receipt.contract_address == Some(self.address) {
+            &transaction.input
+        } else if transaction.to == Some(DEFAULT_CREATE2_DEPLOYER) {
+            &transaction.input[32..]
+        } else {
+            eyre::bail!(
+                "Could not extract the creation code for contract at address {}",
+                self.address
+            );
+        };
+
+        // If bytecode_hash is disabled then its always partial verification
+        let (verification_type, has_metadata) =
+            match (&self.verification_type, config.bytecode_hash) {
+                (VerificationType::Full, BytecodeHash::None) => (VerificationType::Partial, false),
+                (VerificationType::Partial, BytecodeHash::None) => {
+                    (VerificationType::Partial, false)
+                }
+                _ => (VerificationType::Partial, true),
```

Comment:
doesn't this always lead to partial verification, even in the case of `VerificationType::Full` and `BytecodeHash::Some(...)`?